### DISCUSSION TITLE
Opening Positron via SSH and Activating Conda Environment

### DISCUSSION THREAD

**Discussion by laguill** [posit-dev/positron#5740] [Q&A]
Hello,

I’m trying to figure out if it's possible to open Positron (the IDE) from the command line and connect to a remote host via SSH. My goal is to:

Launch Positron from the command line.
Establish an SSH connection to a remote host.
Activate a specific Conda environment on the remote host within Positron.
Does anyone know how to achieve this, or if there’s a straightforward way to configure Positron to automatically connect to an SSH host and activate a Pixi environment?

Thanks in advance for any guidance!

**Comment by nstrayer** [comment]
There is (mostly undocumented) ssh remote host support. See https://github.com/posit-dev/positron/pull/4251. 

However, there may be issues using Pixi (https://github.com/posit-dev/positron/issues/5248). How important is Pixi to this workflow?

**Reply by laguill** [reply]
I am using Pixi as an environment manager. 
Connecting to the remote server via SSH from Positron works fine.

The issue arises during local development when I need Positron to recognize the Quarto package installed with conda in my environment. 

To achieve this locally, I have to execute the following commands: 
```bash
cd project
pixi shell
positron .
``` 
However, I am wondering if it is possible to achieve the same setup when working over SSH. The challenge is that Positron is installed locally, while the Pixi environment resides on the remote server.

**Comment by juliasilge** [comment]
Are you wanting to use a pixi environment for R or Python? If it is for Python, I believe this should "just work", i.e. you can establish an SSH connection to a remote host, and then your pixi Python environment on the remote host should be discovered, can be selected, can start, etc.

Unfortunately, for R, I do not expect this would work right now because of #4398 and #3724; we do not automatically detect these R interpreters.

**Reply by juliasilge** [reply]
We've just merged in docs on remote SSH development at <https://positron.posit.co/remote-ssh> if you'd like to read more.

**Reply by clydeugene** [reply]
Hi @juliasilge 

I have been using a number of `pixi` environments on a remote Ubuntu 20.04 server via `ssh` without any issues until yesterday when I had to move my files to a different drive on the server. After completing the move, I cannot see my interpreters in the `Start New Interpreter Session` dropdown. 


<img width="589" height="2185" alt="CleanShot 2025-10-23 at 11 46 52@2x" src="https://github.com/user-attachments/assets/3b1d65f5-6846-4750-9901-a7ec562c8396" />


Initially, I had the paths to the python binaries defined in [python.interpreters.include](positron://settings/python.interpreters.include) on the remote. After moving my files around, and editing the paths in that setting I can no longer see any of the interpreters. I noticed that I could theoretically supply a top level folder as a catch-all (see screenshot below) but even that did not work.


<img width="2324" height="724" alt="CleanShot 2025-10-23 at 11 53 50@2x" src="https://github.com/user-attachments/assets/66ad357f-3df4-4313-86fa-e9ada9145381" />


Using the built-in `ms-python` extension, I can see the pixi environments, and when I try to launch one, it either opens briefly, before being replaced by another Python session when I open a Python script, or it does not launch at all. I have added a screenshot of the file picker from that extension below:


<img width="1224" height="1136" alt="CleanShot 2025-10-23 at 11 33 31@2x" src="https://github.com/user-attachments/assets/db6aa756-93c4-4649-b37b-093230bcca11" />

This fails on both the Windows and Mac versions and I assume it is because they are synced, and inherit the same config on the server.

Might you have any recommendations?

**Reply by clydeugene** [reply]
I figured out the issue. Copying the working directory to the new location broke hardcoded `conda` links in the `pixi` environments. So, I needed to reinstall the environments to fix the broken links.

Once this was done, I reran the interpreter discovery and Positron found the environments.

Note for my future self - reinstall the environments from the `pixi.toml` file 😅
